lib/DXIL/DxilShaderModel.cpp

```diff
@@ -494,6 +494,20 @@ const llvm::StringRef ShaderModel::FullNameFromKind(DXIL::ShaderKind sk) {
   }
 }
 
+bool ShaderModel::AllowDerivatives(DXIL::ShaderKind sk) const {
+  switch (sk) {
+  case DXIL::ShaderKind::Pixel:
+  case DXIL::ShaderKind::Library:
+    return true;
+  case DXIL::ShaderKind::Compute:
+  case DXIL::ShaderKind::Amplification:
+  case DXIL::ShaderKind::Mesh:
+  case DXIL::ShaderKind::Node:
```

Comment:
While this wouldn't cause a problem, node doesn't need the shader model check, since it only exists in SM 6.8+.  There should be an independent check for trying to use Node in a lower shader model.

Reply:
Updated.